addons/Request/Request.lua

```diff
@@ -0,0 +1,305 @@
+--[[
+Copyright © 2015, Selindrile
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Request nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Selindrile BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+]]
+
+require('luau')
+
+_addon.name = 'Request'
+_addon.author = 'Selindrile'
+_addon.commands = {'request','rq'}
+_addon.version = 1.1
+_addon.language = 'english'
+
+defaults = T{}
+defaults.mode = 'whitelist'
+defaults.whitelist = S{}
+defaults.blacklist = S{}
+defaults.nicknames = S{}
+defaults.forbidden = S{}
+defaults.partylock = 'On'
```

Comment:
You can make a function like this:
```lua
display = function(setting)
    if class(setting) == 'Set' then
        return setting:empty() and '(empty)' or setting:format('csv')
    elseif class(setting) == 'boolean' then
        return setting and 'on' or 'off'
    end

    return tostring(setting)
end
```

You can also adjust your status display function:
```lua
    elseif command == 'status' then
        for key in pairs(defaults) do
            log('%s: %s':format(key, settings[key]))
        end
    end
```

That way you can add new settings at any time and it'll display them correctly without you manually having to add them to the status list.